lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -1035,22 +1295,72 @@ class MongoClientInterface {
+     * @param {Object} params.mongifiedSearch search options (optional)
+     * @param {Object} log logger
+     * @param {Function} cb callback
+     * @return {undefined}
+     */
     listObject(bucketName, params, log, cb) {
-        const extName = params.listingType;
-        const extension = new listAlgos[extName](params, log);
-        const internalParams = extension.genMDParams();
-        internalParams.mongifiedSearch = params.mongifiedSearch;
-        return this.internalListObject(bucketName, internalParams, extension,
-            log, cb);
+        return this.getBucketVFormat(bucketName, log, (err, vFormat) => {
+            if (err) {
+                return cb(err);
+            }
+            const extName = params.listingType;
+            // extention here can either be DelimiterMaster ot DelimiterVersions
+            const extension = new listAlgos[extName](params, log, vFormat);
+            // the params returned depend on the vFormat as well as the algorithm used
+            // DelimiterMaster returns an object of filters to apply when filtering
+            // DelimiterVersions in v0 also returns the same thing
+            // in v1 however it returns two objects containing filters to use for two separate
+            // listing streams (master and version)
+            const extentionParams = extension.genMDParams();
+            const internalParams = {
+                mainStreamParams: Array.isArray(extentionParams) ? extentionParams[0] : extentionParams,
+                secondaryStreamParams: Array.isArray(extentionParams) ? extentionParams[1] : null,
+            };
+            internalParams.mongifiedSearch = params.mongifiedSearch;
+            return this.internalListObject(bucketName, internalParams, extension,
+                log, cb);
+        });
     }
 
+    /**
+     * lists versionned and non versionned objects in a bucket
+     * @param {String} bucketName bucket name
+     * @param {Object} params params
+     * @param {String} params.listingType type of algorithm to use (optional)
+     * @param {Number} params.maxKeys maximum number of keys to list (optional)
+     * @param {String} params.prefix prefix of objects to use (optional)
+     * @param {String} params.delimiter delimiter to use (optional)
+     * @param {Object} params.mongifiedSearch search options (optional)
+     * @param {Object} log logger
+     * @param {Function} cb callback
+     * @return {undefined}
+     */
     listMultipartUploads(bucketName, params, log, cb) {
-        const extName = params.listingType;
-        const extension = new listAlgos[extName](params, log);
-        const internalParams = extension.genMDParams();
-        internalParams.mongifiedSearch = params.mongifiedSearch;
-        return this.internalListObject(bucketName, internalParams, extension,
-            log, cb);
+        return this.getBucketVFormat(bucketName, log, (err, vFormat) => {
```

Comment:
I'm not sure how this works with Mongo, but I'm surprised that we need to take into account the vFormat of the bucket to list its multipart-uploads. Normally (in metadata at least), listing multipart uploads goes through a special sidekick bucket that is just a flat list of open multipart uploads, hence always in v0. Can you check this to make sure it's correct?

Reply:
Well i didn't exclude the mpu buckets in the createBucket function so they too have a v0 and v1 key format, but you're right there's no point in having them support the v1 format, i'll fix that